apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,77 @@
+import { IPreferenceChannels, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepResponseDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      tags: template.tags,
+      active: template.active,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepResponseDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin || WorkflowOriginEnum.NOVU,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      name: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepTypeOverviews: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
+    };
+  }
+
+  static toWorkflowsMinifiedDtos(templates: NotificationTemplateEntity[]): MinifiedResponseWorkflowDto[] {
+    return templates.map(WorkflowTemplateGetMapper.toMinifiedWorkflowDto);
+  }
+  static toStepResponseDto(step: NotificationStepEntity): StepResponseDto {
+    return {
+      name: step.name || 'Missing Name',
+      stepUuid: step._templateId,
+      type: step.template?.type || StepTypeEnum.EMAIL,
+      controls: WorkflowTemplateGetMapper.convertControls(step),
+      active: step.active || true,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** There isn't a concept of "active" steps in Novu V2 - can we remove the `active` attribute?‏

Reply:
removed as a part of @djabarovgeorge PR comments